Cython/Build/Dependencies.py

```diff
@@ -878,7 +878,8 @@ def copy_to_build_dir(filepath, root=root):
                 new_sources.append(source)
                 if build_dir:
                     copy_to_build_dir(source)
-        m.sources = new_sources
+        if not hasattr(options, 'keep_sources') or not options.keep_sources:
+            m.sources = new_sources
```

Comment:
We should probably still add any additional sources that were present, just avoid swapping out the .pyx for .c[pp].